### PR TITLE
Disable multiple buttons for discharged patients

### LINKED ISSUE #7722: Disable multiple buttons for discharged patients
**Describe the bug**

few buttons are still active for discharged patients on the consultation page

**To Reproduce**

**Expected behavior**

image

- [ ] disable log update edit button for discharge patient

image

- [ ] disable all file upload related button in consultation page for patient

image

- [ ] disable edit button in the already existing investigation view details page

### PR DESCRIPTION
**Pull request by alanansari**
## Proposed Changes

- Fixes #7722
- Disabled multiple buttons based on the consultation discharge date for patients
image
image
image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
LGTM

File: src/Utils/request/README.md
Comment (rithviknishad) [inline src/Utils/request/README.md:1]: Could you undo the formatting done to unrelated files such as these?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Patient/FileUpload.tsx [hunk 1]
```diff
@@ -1607,6 +1618,7 @@ export const FileUpload = (props: FileUploadProps) => {
                         }
                       </AuthorizedChild>
                       <ButtonV2
+                        disabled={consultation?.discharge_date ? true : false}
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
                        disabled={!!consultation?.discharge_date}
```

File: src/Components/Patient/FileUpload.tsx [hunk 2]
```diff
@@ -1599,6 +1607,9 @@ export const FileUpload = (props: FileUploadProps) => {
                                 type="file"
                                 accept="image/*,video/*,audio/*,text/plain,text/csv,application/rtf,application/msword,application/vnd.oasis.opendocument.text,application/pdf,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,application/vnd.oasis.opendocument.spreadsheet,application/pdf"
                                 hidden
+                                disabled={
+                                  consultation?.discharge_date ? true : false
+                                }
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
                                disabled={!!consultation?.discharge_date}
```

File: src/Components/Patient/FileUpload.tsx [hunk 3]
```diff
@@ -1586,7 +1592,9 @@ export const FileUpload = (props: FileUploadProps) => {
                       <AuthorizedChild authorizeFor={NonReadOnlyUsers}>
                         {({ isAuthorized }) =>
                           isAuthorized ? (
-                            <label className="button-size-default button-shape-square button-primary-default inline-flex h-min w-full cursor-pointer items-center justify-center gap-2 whitespace-pre font-medium outline-offset-1 transition-all duration-200 ease-in-out disabled:cursor-not-allowed disabled:bg-gray-200 disabled:text-gray-500">
+                            <label
+                              className={`button-size-default button-shape-square inline-flex h-min w-full items-center justify-center gap-2 whitespace-pre font-medium outline-offset-1 ${consultation?.discharge_date ? "cursor-not-allowed bg-gray-200 text-gray-500" : "button-primary-default cursor-pointer transition-all duration-200 ease-in-out"}`}
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: use `classNames` utility method for conditional classes

File: src/Components/Patient/FileUpload.tsx [hunk 4]
```diff
@@ -1526,6 +1531,7 @@ export const FileUpload = (props: FileUploadProps) => {
                     )}
                     <div className="flex flex-col items-center gap-4 md:flex-row md:flex-wrap lg:flex-nowrap">
                       <VoiceRecorder
+                        isDisabled={consultation?.discharge_date ? true : false}
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
                        isDisabled={!!consultation?.discharge_date}
```

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 5]
```diff
@@ -151,6 +156,7 @@ export default function ShowInvestigation(props: any) {
       <InvestigationTable
         title={`ID: ${sessionId}`}
         data={state.initialValues}
+        isDischargedPatient={consultation?.discharge_date ? true : false}
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx]:
```suggestion
        isDischargedPatient={!!consultation?.discharge_date}
```

File: src/Components/Facility/Investigations/ShowInvestigation.tsx [hunk 6]
```diff
@@ -80,6 +80,11 @@ export default function ShowInvestigation(props: any) {
     },
   );
 
+  const { data: consultation } = useQuery(routes.getConsultation, {
+    pathParams: { id: consultationId },
+    prefetch: !!consultationId,
+  });
```
Comment (rithviknishad) [inline src/Components/Facility/Investigations/ShowInvestigation.tsx:86]: Can we pass the consultation object as prop to this component instead of making API call just for checking whether to disable or not?

File: src/Components/Patient/FileUpload.tsx [hunk 7]
```diff
@@ -258,6 +258,11 @@ export const FileUpload = (props: FileUploadProps) => {
     prefetch: !!patientId,
   });
 
+  const { data: consultation } = useQuery(routes.getConsultation, {
+    pathParams: { id: consultationId },
+    prefetch: !!consultationId,
+  });
+
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx:265]: Can we pass the consultation object as an optional prop to this component instead of making API call just for checking whether to disable or not?

File: src/Components/Facility/Consultations/DailyRounds/DefaultLogUpdateCard.tsx [hunk 8]
```diff
@@ -49,6 +49,7 @@ const DefaultLogUpdateCard = ({ round, ...props }: Props) => {
         </ButtonV2>
         <ButtonV2
           variant="secondary"
+          disabled={props.consultationData?.discharge_date ? true : false}
```
Comment (rithviknishad) [inline src/Components/Facility/Consultations/DailyRounds/DefaultLogUpdateCard.tsx]:
```suggestion
          disabled={!!props.consultationData?.discharge_date}
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Patient/FileUpload.tsx (hunk 7 above)
Comment (alanansari) [reply src/Components/Patient/FileUpload.tsx:265]: There's no consultation object available at that level to pass as prop.
The same thing was done for patient object in the component.

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]
